include/util/shared_memory_vector_wrapper.hpp

```diff
@@ -126,20 +139,21 @@ template <typename DataT> class SharedMemoryWrapper
 
     auto data() const { return m_ptr; }
 
-    template <typename T>
-    friend void swap(SharedMemoryWrapper<T> &, SharedMemoryWrapper<T> &) noexcept;
+    template <typename T> friend void swap(vector_view<T> &, vector_view<T> &) noexcept;
 };
 
-template <> class SharedMemoryWrapper<bool>
+template <> class vector_view<bool>
 {
   private:
     unsigned *m_ptr;
     std::size_t m_size;
 
   public:
-    SharedMemoryWrapper() : m_ptr(nullptr), m_size(0) {}
+    using value_type = bool;
 
-    SharedMemoryWrapper(unsigned *ptr, std::size_t size) : m_ptr(ptr), m_size(size) {}
+    vector_view() : m_ptr(nullptr), m_size(0) {}
+
+    vector_view(unsigned *ptr, std::size_t size) : m_ptr(ptr), m_size(size) {}
```

Comment:
Here and in reset: should the unsigned ptrs be const? the vector view is not able to mutate after all. Also the member m_ptr should probably be a `const unsigned *` (read-only data, but ptr can be swapped out, e.g. inreset)